---
title: "CourseInfo"
format: 
  html:
    theme: 
      light: flatly
      dark: darkly
---

```{r}
#| include: false
knitr::opts_chunk$set(echo = FALSE)

library(knitr)
library(magrittr)
library(dplyr)
```

# Welcome to BIFX 552 - Bioinformatics Data Skills

## Syllabus

### Course Description

This class provides an introduction to applied data science skills needed by bioinformatics professionals. A focus will be placed on reproducible bioinformatics research and will include the following topics and tools: beginning to intermediate use of the Unix command line, working with remote computing resources, version tracking, R and Bioconductor, tools for manipulating sequence data, and creation of pipelines.

-   Instructor: Randall Johnson, PhD
-   Office hours: Office hours will be held immediately before and after in-person meetings. Drop me a note to schedule a virtual office hour at another time.
-   Prerequisites: BIFX 503
-   Textbooks: We will not be using a textbook this term, but you may find the following helpful. All are freely available online.
    -   [R for Data Science](https://r4ds.had.co.nz/)
    -   [Advanced R](https://adv-r.hadley.nz/)
    -   [R Packages](https://r-pkgs.org/)
-   Communications: Course communications will be posted on [Blackboard](https://hood.blackboard.com).
-   Code of Conduct: We want to foster a safe, enjoyable and productive learning environment. People like you add value to our program. To meet this end, all participants will be expected to follow the [course code of conduct](code_of_conduct.html) located in the course documents on Blackboard.
-   [GradCare](https://www.hood.edu/hood-community/hood-college-health-wellness/gradcare): Graduate school can be stressful. Hood College is here to help. We offer a wide range of support services for graduate students. Our programs and resources address the eight dimensions of wellness—emotional, physical, social, intellectual, spiritual, occupational, financial and environmental—so you can stay healthy while maintaining school-work-life balance.
-   Accessibility: As the instructor of this course, I endeavor to provide an inclusive learning environment. However, if you experience barriers to learning in this course, do not hesitate to discuss them with me and the Office for Accessibility Services, 301-696-3421, gmuer\@hood.edu.

### Learning Objectives

On completion of this course, students should be comfortable with the following:

-   Use of R and Bioconductor to perform bioinformatic analysis tasks
-   Use of the Unix command line to manipulate data and perform bioinformatic analysis tasks
-   Logging into and using remote computing resources
-   Working with version controlled code repositories in a collaborative work environment
-   Stitching a series of commands and/or programs together into a reusable pipeline

### Asynchronous and In-Person Content

Asynchronous content will be posted weekly, and in-person labs will be scheduled on Thursday evenings throughout the term at 5:30 PM (see the schedule below). Course content will be based on up-to-date, publicly available material from experts in the [Data Carpentry](https://datacarpentry.org/), [Software Carpentry](https://software-carpentry.org/), and [Bioconductor](https://bioc2023.bioconductor.org/) communities.

### Homework and Course Project

There will be homework assignments associated with each course module. This will give you an opportunity to practice what we cover each week. Homework from asynchronous content will be due on Thursdays and should be completed before each in-person meeting. The last few weeks of the course will be devoted to a group project based on a bioinformatics workflow from Bioconductor.

### Grading

Grades will be based on completion of homework, a group project and two exams.

-   Asynchronous Content - 25%
-   In-person Content - 25%
-   Bioconductor Project - 10%
-   Mid-term - 20%
-   Final exam - 20%

Late submissions of any work will be subject to a 5% penalty for each day past the due date. The maximum penalty for late work will be 50%.

### Weather

In the event of severe weather resulting in the closure of Hood College and the cancellation of an in-person meeting, the meeting will be held virtually at the scheduled time and a recording posted on Blackboard for asynchronous viewing.

### Tentative Class Schedule

```{r schedule}
#### Git and GitHub ####
schedule <- 
    tibble(Week = 1,
           Date = 'Aug 31', 
           `Coding Demo` =       'Git',
           Lab           = paste('[Class intro](https://bifx552-23.github.io/CourseInfo/),',
                                 '[Markdown / Quarto](labs/01markdown)'))

#### R for Reproducible Research ####
gapminder <- 'https://swcarpentry.github.io/r-novice-gapminder/'
bioc_intro <- 'https://carpentries-incubator.github.io/bioc-intro/'

schedule <- bind_rows(schedule,
        tibble(Week = 2,
               Date = 'Sep 7', 
               `Coding Demo` = 'R Intro',
               Lab           = '*[git practice](https://learngitbranching.js.org) (virtual)*'))

schedule <- bind_rows(schedule,
        tibble(Week = 3,
               Date = 'Sep 14', 
               `Coding Demo` = 'Subsetting Data, R Control Flow',
               Lab           = '[P-Hacking](labs/02phacking)'))

schedule <- bind_rows(schedule,
        tibble(Week = 4,
               Date = 'Sep 21', 
               `Coding Demo` = 'R Graphics, R Vectorization, R Functions',
               Lab           = 'COVID Data'))

schedule <- bind_rows(schedule,
        tibble(Week = 5,
               Date = 'Sep 28', 
               `Coding Demo` = 'R Advanced Data',
               Lab           = 'RNA-Seq Data'))

schedule <- bind_rows(schedule,
        tibble(Week = 6,
               Date = 'Oct 5', 
               `Coding Demo` = 'Joining Tables, R Software Tips',
               Lab           = '*Exam review (virtual)*'))

#### Midterm ####

schedule <- bind_rows(schedule,
        tibble(Week = 7,
               Date = 'Oct 12', 
               `Coding Demo` = '*Midterm*',
               Lab           = "*Exam Due (no class)*"))

#### UNIX and Snakemake ####
shell_genomics <- 'https://datacarpentry.org/shell-genomics/'

schedule <- bind_rows(schedule,
        tibble(Week = 8,
               Date = 'Oct 19', 
               `Coding Demo` = 'Unix',
               Lab           = 'Unix lab?'))
# Lab ideas:
#  * make login script
#  * write script to process a bunch of files - batch convert images using imageMagick?

schedule <- bind_rows(schedule,
        tibble(Week = 9,
               Date = 'Oct 26', 
               `Coding Demo` = 'Snakemake',
               Lab           = '*Linux practice (virtual)*'))

#### Variant Calling and Snakemake ####

schedule <- bind_rows(schedule,
        tibble(Week = 10,
               Date = 'Nov 2', 
               `Coding Demo` = 'Variant Calling Workflow',
               Lab           = 'Variant Calling Pipeline'))
# do this over two weeks with https://carpentries-incubator.github.io/snakemake-novice-bioinformatics/05-expansion/index.html in-between, perhaps cover some of https://carpentries-incubator.github.io/snakemake-novice-bioinformatics/06-awkward_programs/index.html between parts 1 and 2 as well.

schedule <- bind_rows(schedule,
        tibble(Week = 11,
               Date = 'Nov 9', 
               `Coding Demo` = 'Advanced Snakemake', # (5, 7 - 9) cover 6 in lab?
               Lab           = 'Variant Calling Pipeline (cont)'))

#### Bioconductor ####

schedule <- bind_rows(schedule,
        tibble(Week = 12,
               Date = 'Nov 16', 
               `Coding Demo` = 'Intro to Bioconductor',
               Lab           = 'Bioconductor Project'))
# review Bioconductor project site
# review Biostrings vignette(s) https://www.bioconductor.org/packages/release/bioc/html/Biostrings.html

schedule <- bind_rows(schedule,
        tibble(Week = 13,
               Date = 'Nov 23', 
               `Coding Demo` = '*Group Work*',
               Lab           = '*Thanksgiving (no class)*'))

schedule <- bind_rows(schedule,
        tibble(Week = 14,
               Date = 'Nov 30', 
               `Coding Demo` = '*Group Work*',
               Lab           = 'Project Presentations'))

schedule <- bind_rows(schedule,
        tibble(Week = 15,
               Date = 'Dec 7',
               `Coding Demo` = '*Final*',
               Lab           = '*Exam Due (no class)*'))

rename(schedule,
       `Asynchronous Content` = `Coding Demo`,
       `In Person Content` = Lab) %>%
    kable()
```
