---
title: "CourseInfo"
format: 
  html:
    theme: 
      light: flatly
      dark: darkly
---

```{r}
#| include: false
knitr::opts_chunk$set(echo = FALSE)

library(knitr)
library(magrittr)
library(dplyr)
```

# Welcome to BIFX 553 - Bioinformatics Data Skills

## Syllabus

### Course Description

This class provides an introduction to applied data science skills needed by bioinformatics professionals. A focus will be placed on reproducible bioinformatics research and will include the following topics and tools: beginning to intermediate use of the Unix command line, working with remote computing resources, version tracking, R and Bioconductor, tools for manipulating sequence data, and creation of pipelines.

-   Instructor: Randall Johnson, PhD
-   Office hours: Office hours will be held immediately before and after labs. Drop me a note to schedule a virtual office hour at another time.
-   Prerequisites: BIFX 503
-   Textbooks: We will not be using a textbook this term, but you may find the following helpful. All are freely available online.
    -   [R for Data Science](https://r4ds.had.co.nz/)
    -   [Advanced R](https://adv-r.hadley.nz/)
    -   [R Packages](https://r-pkgs.org/)
-   Communications: Course communications will be posted on [Blackboard](https://hood.blackboard.com).
-   Code of Conduct: We want to foster a safe, enjoyable and productive learning environment. People like you add value to our program. To meet this end, all participants will be expected to follow the [course code of conduct](code_of_conduct.html) located in the course documents on Blackboard.
-   Accessibility: As the instructor of this course, I endeavor to provide an inclusive learning environment. However, if you experience barriers to learning in this course, do not hesitate to discuss them with me and the Office for Accessibility Services, 301-696-3421, gmuer\@hood.edu.

### Learning Objectives

On completion of this course, students should be comfortable with the following:

-   Use of R and Bioconductor to perform bioinformatic analysis tasks
-   Use of the Unix command line to manipulate data and perform bioinformatic analysis tasks
-   Logging into and using remote computing resources
-   Working with version controlled code repositories in a collaborative work environment
-   Stitching a series of commands and/or programs together into a reusable pipeline

### Coding Demos and In-Class Labs

Recorded coding demos will be posted weekly, and in-class coding labs will be scheduled on Thursday evenings throughout the term at 5:30 PM (unless otherwise specified). Because Bioinformatics is such a fast-moving field, demos and labs will be based on up-to-date, publicly available material from experts in the [Data Carpentry](https://datacarpentry.org/), [Software Carpentry](https://software-carpentry.org/), and [Bioconductor](https://bioc2023.bioconductor.org/) communities.

### Homework and Course Project

There will be homework assignments based on each of the weekly coding demos. This will give you an opportunity to practice what we cover each week. Homework should be completed before class each week.

Students will develop a repository with examples of the material and techniques we will be covering throughout the course. The last few weeks of the course will be devoted to a group project based on a bioinformatics workflow from Bioconductor.

### Grading

Grades will be based on completion of homework, quizzes and two exams.

-   Homework - 25%
-   Labs - 25%
-   Bioconductor Project - 10%
-   Mid-term - 20%
-   Final exam - 20%

Late submissions of any work will be subject to a 5% penalty for each day past the due date. The maximum penalty for late work will be 50%.

### Weather

In the event of severe weather resulting in the closure of Hood College and the cancellation of an in-person lab, the lab will be held virtually at the scheduled time and a recording posted on Blackboard for asynchronous viewing.

### Tentative Class Schedule

```{r schedule}
#### Git and GitHub ####
schedule <- 
    tibble(Week = 1,
           Date = 'Aug 24', 
           `Coding Demo` =       '[Git](https://swcarpentry.github.io/git-novice/) (1 - 9)',
           Lab           = paste('[Class intro](https://bifx552-23.github.io/CourseInfo/),',
                                 '[Markdown / Quarto](labs/01markdown)'))

#### R for Reproducible Research ####
gapminder <- 'https://swcarpentry.github.io/r-novice-gapminder/'
bioc_intro <- 'https://carpentries-incubator.github.io/bioc-intro/'

schedule <- bind_rows(schedule,
        tibble(Week = 2,
               Date = 'Aug 31', 
               `Coding Demo` = paste0('[R Intro](', gapminder, ') (1-3)'),
               Lab           = ''))

schedule <- bind_rows(schedule,
        tibble(Week = 3,
               Date = 'Sep 7', 
               `Coding Demo` = paste0('[Subsetting Data](', gapminder, '06-data-subsetting/index.html), ',
                                      '[R Control Flow](',  gapminder, '07-control-flow/index.html)'),
               Lab           =        '[P-Hacking](labs/02phacking)'))

schedule <- bind_rows(schedule,
        tibble(Week = 4,
               Date = 'Sep 14', 
               `Coding Demo` = paste0('[R Graphics](',      gapminder, '08-plot-ggplot2/index.html), ',
                                      '[R Vectorization](', gapminder, '09-vectorization/index.html), ',
                                      '[R Functions](',     gapminder, '10-functions/index.html)'),
               Lab           =        'COVID Data'))

schedule <- bind_rows(schedule,
        tibble(Week = 5,
               Date = 'Sep 21', 
               `Coding Demo` = paste0('[R Advanced Data](', gapminder, ') (11, 13-14)'),
               Lab           =        'RNA-Seq Data'))

schedule <- bind_rows(schedule,
        tibble(Week = 6,
               Date = 'Sep 28', 
               `Coding Demo` = paste0('[Joining Tables](', bioc_intro, '50-joining-tables/index.html), ',
                                      '[R Software Tips](', gapminder, '16-wrap-up/index.html)'),
               Lab           =        'DockerHub / Shiny'))

#### Midterm ####

schedule <- bind_rows(schedule,
        tibble(Week = 7,
               Date = 'Oct 5', 
               `Coding Demo` = '',
               Lab           = "Midterm Exam Due"))

#### UNIX and Snakemake ####
shell_genomics <- 'https://datacarpentry.org/shell-genomics/'

schedule <- bind_rows(schedule,
        tibble(Week = 8,
               Date = 'Oct 12', 
               `Coding Demo` = paste0('[Unix](',        shell_genomics, ') (1 - 4)'),
               Lab           = 'Unix lab'))
# Lab ideas:
#  * make login script
#  * write script to process a bunch of files - batch convert images using imageMagick?

schedule <- bind_rows(schedule,
        tibble(Week = 9,
               Date = 'Oct 19', 
               `Coding Demo` = '[Snakemake](https://carpentries-incubator.github.io/snakemake-novice-bioinformatics/) (1 - 4)',
               Lab           = 'Snakemake Intro'))

#### Variant Calling and Snakemake ####

schedule <- bind_rows(schedule,
        tibble(Week = 10,
               Date = 'Oct 26', 
               `Coding Demo` = paste0('[Variant Calling Workflow](https://datacarpentry.org/wrangling-genomics/) (1 - 4)'),
               Lab           = 'Variant Calling Pipeline'))
# do this over two weeks with https://carpentries-incubator.github.io/snakemake-novice-bioinformatics/05-expansion/index.html in-between, perhaps cover some of https://carpentries-incubator.github.io/snakemake-novice-bioinformatics/06-awkward_programs/index.html between parts 1 and 2 as well.

schedule <- bind_rows(schedule,
        tibble(Week = 11,
               Date = 'Nov 2', 
               `Coding Demo` = '[Advanced Snakemake](https://carpentries-incubator.github.io/snakemake-novice-bioinformatics/)', # (5, 7 - 9) cover 6 in lab
               Lab           = 'Variant Calling Pipeline (cont)'))

#### Bioconductor ####

schedule <- bind_rows(schedule,
        tibble(Week = 12,
               Date = 'Nov 9', 
               `Coding Demo` = paste0('[Intro to Bioconductor](https://carpentries-incubator.github.io/bioc-project/index.html)(1 - 4)'),
               Lab           = 'Bioconductor Project'))
# review Bioconductor project site
# review Biostrings vignette(s) https://www.bioconductor.org/packages/release/bioc/html/Biostrings.html

schedule <- bind_rows(schedule,
        tibble(Week = 13:14,
               Date = paste('Nov', c(16, 23)), 
               `Coding Demo` = '',
               Lab           = c('Bioconductor Project', 'Thanksgiving (no class)')))

schedule <- bind_rows(schedule,
        tibble(Week = 15,
               Date = 'Nov 30', 
               `Coding Demo` = '',
               Lab           = 'Project Presentations'))

schedule <- bind_rows(schedule,
        tibble(Week = 16,
               Date = 'Dec 7',
               `Coding Demo` = '',
               Lab           = 'Exam Due (no class)'))


kable(schedule)
```
